File: 101_Symmetric Tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
 //Recursion
var isSymmetric = function(root) {
    return help(root, root);
};

const help = (t1, t2) => {
    if(!t1 && !t2) return true;
    if(!t1 || !t2) return false;
    return (t1.val === t2.val) && help(t1.left, t2.right) && help(t1.right, t2.left);
    
}

//Iterative
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @return {boolean}
 */
var isSymmetric = function(root) {
    let arr = []
    arr.push(root);
    arr.push(root);
    
    while(arr.length) {
        let t1 = arr.shift();
        let t2 = arr.shift();
        
        if(!t1 && !t2) continue;
        if(!t1 || !t2) return false;
        if(t1.val !== t2.val) return false;
        
        arr.push(t1.left);
        arr.push(t2.right);
        arr.push(t1.right);
        arr.push(t2.left);
    }
    return true;
};

File: 101_Symmetric Tree.go
/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    queue := []*TreeNode {}
    
    queue = append(queue, root)
    queue = append(queue, root)
    
    for len(queue) > 0 {
        t1 := queue[0]
        queue = queue[1:]
        t2 := queue[0]
        queue = queue[1:]
        
        if t1 == nil && t2 == nil {
            continue
        }
        if t1 == nil || t2 == nil {
            return false
        }
        
        if t1.Val != t2.Val {
            return false
        }
        queue = append(queue, t1.Left)
        queue = append(queue, t2.Right)
        queue = append(queue, t1.Right)
        queue = append(queue, t2.Left)
    }
    return true
}

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */
func isSymmetric(root *TreeNode) bool {
    return isMirror(root, root)
}

func isMirror(t1, t2 *TreeNode) bool {
    if t1 == nil && t2 == nil {
        return true
    }
    if t1 == nil || t2 == nil {
        return false
    }
    
    return (t1.Val == t2.Val) && isMirror(t1.Right, t2.Left) && isMirror(t1.Left, t2.Right)
}